rdm/0004-installation-and-upgrade-workflow.md

```diff
@@ -0,0 +1,161 @@
+
+The *development* environment would run the application using Flask's debug server and connect to the cache, database, queueing and elasticsearch services running in their corresponding containers. In this case both UI and REST applications would run in the same application.
+
+The *semi-production* environment is more complex. It runs the web UI and RESTful API in separate containers, and exposes them by using a Nginx frontend. In this scenario both the base and application images must be built. However, in cases such as updating the application the dependencies might not change, and therefore there is no need to rebuild the base image. To make this possible the flags `--base` and `--app` are used.
+
+Note that the `build` command only builds the images, it does not start the containers.
+
+**setup**
+
+The `setup` command is required to be run only once, upon instance creation. This command sets the environment:
+
+- Destroy the database if existing.
+- Create the database and initialize it with the needed tables.
+- Destroy indices if existing.
+- Create the indices.
+- Purge the queue.
+- Create a default location for files.
+- Create an admin role in order to restrict access.
+
+**run**
+
+The `run` command boots up the whole system. It starts the services containers, and depending on the environment the application containers (*semi-production*) or debug server (*development*).
+
+**update**
+*Needed? See unresolved questions*
+The `update` command allows users to update their instances, in terms of configuration or customizations. This would require the rebuild of the application image.
+
+**upgrade**
+
+The `upgrade` command allows users to upgrade their instances, in terms of Invenio RDM versions. This will require the rebuild of the application image, and potentially the base image. Since this might also carry significant changes, e.g. in the datamodel, the Elasticsearch and Database services would need to be updated.
+
+**cleanup**
+*Needed? See unresolved questions*
+
+**destroy**
+
+The `destroy` command results in the removal of all data, volumnes, containers and application files generated by the `build` and `run` command. Nonetheless, the file structure and contents generated by `init` will remain.
+
+## Usage
+
+In order to use the CLI, it needs to be installed. It is distributed as a package and can be installed via `pip`.
+
+``` bash
+$ pip install invenio-cli
+```
+
+This package will install Cookiecutter, PyDocker and Pipenv among others.
+
+It has a series of hard dependencies that should be installed by the user:
+
+- Python 3.4+
+- Docker
+- Docker Compose v2.x
+
+An example execution to deploy a development instance would be:
+
+``` bash
+$ invenio-cli rdm init
```

Comment:
This should be amended. Due to implementation constraints it has to be:
``` console
invenio-cli --flavour=rdm init
```

The rest is not impacted.